database/migrations/000012_rule_evaluations_rule_hash.up.sql

```diff
@@ -0,0 +1,25 @@
+-- Copyright 2024 Stacklok, Inc
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--      http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+-- Add pull_request_id to rule_evaluations
+ALTER TABLE rule_evaluations
+    ADD COLUMN rule_hash TEXT NOT NULL;
```

Comment:
I don't think you'll be able to do this. The `NOT NULL` constraint will come into effect on existing rules in staging/prod and will fail to execute.

I'd suggest this being a three-step process:

* Add the new column without the `NOT NULL` constraint.
* Iterate existing rule evaluations in the db to update the hash (`SELECT FOR UPDATE` is your friend).
* Add the `NOT NULL` constraint.

To test:

* Spawn a local minder instance from `main`.
* Create some sample rules and profiles
* Apply the upgrades (migration + minder restart)
* Verify that the hashes have been added.

If your workflow is based on `make run-docker` and if you want to easily test an upgrade, set the following environment variable:

```
export RUN_DOCKER_NO_TEARDOWN=true
```

This removes the teardown from the `make run-docker` invocation, getting docker to restart only the containers that need rebuilding.